docs/tpm_lpc_spi_interface.md

```diff
@@ -0,0 +1,141 @@
+  - TPM uses SPI mode 0 (CPHA=0, CPOL=0), no other modes are supported.
+  - SPI packet is outlined in Table 48
+  - Data transfer starts from the most significant bit (MSB) and ends on the
+    least significant bit (LSB)
+  ![](images/tpm2_spi_protocol.png)
+  - The first bit transferred determines whether transaction is read or write.
+  - This is followed by 6 bit length from 1 byte to 64 bytes. This is length of
+    the data payload, excludiong 32-bit header.
+  - Length is followed by 24-bit register address, the highest byte is always
+    0xD4. See Table 9 for details how TPM MMIO space maps to register address
+    emitted on the bus.
+    ![](images/tpm2_locality_address.png)
+
+## TPM register interface
+
+According to section 6.3.1 Host may attempt to read less bytes than there
+actually is in register, TPM must handle this, it must also handle read from the
+middle of register, e.g. if we have a 32-bit register at 0x20, 1 byte read from
+0x20 must yield the first byte of that register, read from 0x21 must yield the
+second byte from that register.
+
+Software must never attempt to cross register size in a same transfer. In case
+of LPC this won't be noticed by TPM as all multi-byte LPC transfers are split
+into multiple transfers, e.g. if hosts writes 0x11223344 to register 0x10 on LPC
+it will appear as
+- 0x11 write to 0x10
+- 0x22 write to 0x11
+- 0x33 write to 0x12
+- 0x44 write to 0x13
+
+Even if register boundary were crossed this doesn't make any difference for TPM.
+In case of SPI data payload of a single transfer cannot be bigger than size of
+register and cannot cross register boundary.
+TBD: how should this be handled? TPM specification says:
+> For SPI, if a TPM receives an access request with a length that exceeds the
+> size of the register specified in the transaction address:
+> (...)
+> b. Writes:
+>   The TPM SHALL update the register designated by the start address.
+>   The TPM MAY update additional, adjacent registers within the targeted
+>   address range.
+>   The TPM SHALL NOT change the state of adjacent registers if the writes to
+>   that register are dropped.
+>
+> The TPM SHOULD NOT abort (as defined in Section 6.5.1.1 Bus Aborts) an entire
+> transaction that crosses a register boundary.
+
+
+TPM must implement the following registers (non-exauhstive list):
+- TPM_DID_VID_x - contains Vendor and Device ID of the TPM, see **Table 13** for
+  details.
+- TPM_RID_x - contains revision ID, see **Table 14 for details**.
+- TPM_INTERFACE_ID_x - mandatory register, see **Table 15**.
+- interrupt control/status registers
+- other registers are required, **Table 19** for FIFO interface requirements.
+
+
+## TPM commands
+
+There are two interfaces for sending TPM commands and reading back responses,
+one is CRB not covered here and another FIFO. See section 6.5.2 for details
+about FIFO mode.
+
+In FIFO mode command is written to `TPM_DATA_FIFO_x` register. Host writes
```

Comment:
There is also `TPM_XDATA_FIFO_x` that can be used for larger data transfers, especially useful for SPI.